### PR TITLE
Cherry pick commits from main [9/25-10/21]

### PR DESCRIPTION
**Pull request by mhatrevi**

### CODE REVIEW

File: rom/dev/tests/rom_integration_tests/test_mailbox_errors.rs [hunk 1]
```diff
@@ -151,3 +151,27 @@ fn test_mailbox_invalid_req_size_zero() {
...
+
+#[test]
+// Changing PAUSER not supported on sw emulator
+#[cfg(any(feature = "verilator", feature = "fpga_realtime"))]
+fn test_mailbox_reserved_pauser() {
+    let (mut hw, _image_bundle) =
+        helpers::build_hw_model_and_image_bundle(Fuses::default(), ImageOptions::default());
+
+    // Set pauser to the reserved value
+    hw.set_apb_pauser(0xffffffff);
```
Comment (ArthurHeymans) [inline rom/dev/tests/rom_integration_tests/test_mailbox_errors.rs:163]: set_axi_id

**Review by ArthurHeymans — APPROVED** [review]

**Review by swenson — APPROVED** [review]
LGTM

**Review by rusty1968 — APPROVED** [review]

File: error/src/lib.rs [hunk 2]
```diff
@@ -500,6 +501,7 @@ impl CaliptraError {
     pub const FW_PROC_MAILBOX_PROCESS_FAILURE: CaliptraError = CaliptraError::new_const(0x01020007);
     pub const FW_PROC_MAILBOX_STASH_MEASUREMENT_MAX_LIMIT: CaliptraError =
         CaliptraError::new_const(0x01020008);
+    pub const FW_PROC_MAILBOX_RESERVED_PAUSER: CaliptraError = CaliptraError::new_const(0x01020009);
```
Comment (nquarton) [inline error/src/lib.rs:504]: We have the potential for some divergence on these error codes if we add them in each branch at the same time. Are we okay with the same error potentially having a slightly different code on 1.x vs 2.0? Or do we need to ensure they are consistent?

**Review by nquarton — APPROVED** [review]

**Review by jhand2 — APPROVED** [review]

**Review by zhalvorsen — APPROVED** [review]

File: error/src/lib.rs (hunk 2 above)
Comment (korran) [reply error/src/lib.rs:504]: As I've mentioned before, branching many of these crates is a mistake.

**Comment by korran** [comment]
Are we sure we want to do manual cherry-picks? If we're really going to do this sort of thing, maybe we should turn off "linear history" and then use standard Git merges from main into this branch. At least then it can be tracked automatically...

File: error/src/lib.rs (hunk 2 above)
Comment (mhatrevi) [reply error/src/lib.rs:504]: @nquarton We'll can do a sweep on main-2.x at a later point for the error codes and ensure they are in sync with main.

**Pull request merged**